### PR TITLE
Feature/add btn switch settings

### PR DESCRIPTION
**Pull request by jedi7**
* add configuration tab for GUI into settings
* add configuration checkbox for using left mouse button for dragging and changing the center frequency on FFT spectrum panel.

### CODE REVIEW

**Comment by jedi7** [comment]
please look at this approach if it is ok. There is issue it is not saving the variables. I do not know why the serialize in app is not called. Or where is placed the serialized config file.

**Comment by jedi7** [comment]
ok it seems it is working on app start/exit. But the change is not propagated during run.

File: App/GuiConfig.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  this->deserialize(conf);
+}
+
+void
+GuiConfig::loadDefaults(void)
+{
+  this->useLMBdrag = false;
+}
+
+#define STRINGFY(x) #x
```
Comment (BatchDrake) [inline App/GuiConfig.cpp:40]: There is already a `STRINGIFY` (with an I between STRING and FY, I know, it's odd) with identical behavior you can use by including `<Suscan/Compat.h>`

File: App/GuiConfig.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
+//
+//    GuiConfig.cpp: Gui configuration object
+//    Copyright (C) 2019 Gonzalo José Carracedo Carballal
```
Comment (BatchDrake) [inline App/GuiConfig.cpp:3]: It is okay if you put your name here. You wrote this file after all :)

File: include/GuiConfig.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#ifndef GUICONFIG_H
```
Comment (BatchDrake) [inline include/GuiConfig.h:1]: Add copyright + license header.

File: ui/Config.ui [hunk 4]
```diff
@@ -48,7 +48,7 @@
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
-      <number>0</number>
+      <number>2</number>
```
Comment (BatchDrake) [inline ui/Config.ui:51]: This makes the color config tab the active tab the first time you open the config dialog (you may have changed this accidentally by browsing the config dialog tabs in QtCreator). Just set it back to zero and that would be it.

File: ui/Config.ui [hunk 5]
```diff
@@ -1103,20 +1103,47 @@
...
      </widget>
+     <widget class="QWidget" name="tab_gui">
+      <attribute name="title">
+       <string>GUI behaviour</string>
+      </attribute>
+      <layout class="QGridLayout" name="gridLayout_13">
+       <item row="0" column="0">
+        <widget class="QCheckBox" name="reverseDragBehaviorCheck">
+         <property name="text">
+          <string>Use Left mouse button to drag and change center frequency</string>
```
Comment (BatchDrake) [inline ui/Config.ui:1114]: Minor: put "Left" in lower case, and prefix it with & ("Use &left mouse button...") so you can toggle this option using a keyboard accelerator (Alt + l or equivalent).

**Review by BatchDrake — CHANGES_REQUESTED** [review]
I like these changes. I've just highlighted a few minors I'd like you to change, but apart from that, this would be ready to merge :)

File: App/GuiConfig.cpp (hunk 1 above)
Comment (jedi7) [reply App/GuiConfig.cpp:40]: actually it was forgotten copy/paste from the ColorConfig.cpp ;-)
I was thinking if a template will be helpful here, but unfortunately the stringify must be still done by macro.

**Comment by jedi7** [comment]
Thanks for comments, it is pleasure to co-work with you :)

**Pull request merged**